src/System.IO.Pipelines/System/IO/Pipelines/ReadableBuffer.cs

```diff
@@ -201,18 +221,50 @@ public BufferEnumerator GetEnumerator()
             return new BufferEnumerator(BufferStart, BufferEnd);
         }
 
+        public ReadCursor Move(ReadCursor cursor, long count)
+        {
+            if (count < 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(count));
+            }
+            return cursor.Seek(count, BufferEnd, false);
+        }
+
         /// <summary>
-        /// Create a <see cref="ReadableBuffer"/> over an array.
+        /// Create a <see cref="ReadableBuffer"/> over an <see cref="IEnumerable{OwnedMemory{Byte}}"/>.
         /// </summary>
-        public static ReadableBuffer Create(byte[] data)
+        public static PreservedBuffer Create(IEnumerable<OwnedMemory<byte>> data)
```

Comment:
Can we move the test only APIs to some extensions library?

Reply:
It was an exercise in creating RB for any linked list without before we decided to unify on `IMemoryList`. Will remove.